packages/google-pay-integration/src/google-pay-stripe/google-pay-stripe-gateway.ts

```diff
@@ -0,0 +1,157 @@
+import { includes, some } from 'lodash';
+
+import {
+    InvalidArgumentError,
+    isRequestError,
+    PaymentArgumentInvalidError,
+    PaymentIntegrationSelectors,
+    PaymentIntegrationService,
+    PaymentMethodCancelledError,
+    PaymentMethodFailedError,
+    RequestError,
+} from '@bigcommerce/checkout-sdk/payment-integration-api';
+
+import GooglePayGateway from '../gateways/google-pay-gateway';
+import assertsIsGooglePayStripeInitializationData from '../guards/is-google-pay-stripe-initialization-data';
+import isGooglePayStripeRequestError from '../guards/is-google-pay-stripe-request-error';
+import assertIsGooglePayStripeTokenObject from '../guards/is-google-pay-stripe-token-object';
+import {
+    GooglePayCardDataResponse,
+    GooglePaySetExternalCheckoutData,
+    GooglePayStripeGatewayParameters,
+} from '../types';
+
+import StripeUPEScriptLoader from './stripe-upe-script-loader';
+import { StripeError, StripeUPEClient } from './types';
+
+export default class GooglePayStripeGateway extends GooglePayGateway {
+    private stripeUPEClient?: StripeUPEClient;
+
+    constructor(
+        private paymentIntegrationService: PaymentIntegrationService,
+        private scriptLoader: StripeUPEScriptLoader,
+    ) {
+        super('stripe', paymentIntegrationService);
+    }
+
+    async mapToExternalCheckoutData(
+        response: GooglePayCardDataResponse,
+    ): Promise<GooglePaySetExternalCheckoutData> {
+        const data = await super.mapToExternalCheckoutData(response);
+
+        let token: unknown;
+
+        try {
+            token = JSON.parse(data.nonce);
+        } catch (error) {
+            throw new InvalidArgumentError('Unable to parse response from Google Pay.');
+        }
+
+        assertIsGooglePayStripeTokenObject(token);
+
+        data.nonce = token.id;
+
+        return data;
+    }
+
+    getPaymentGatewayParameters(): GooglePayStripeGatewayParameters {
+        const data = this.getGooglePayInitializationData();
+
+        assertsIsGooglePayStripeInitializationData(data);
+
+        const { stripeVersion, stripePublishableKey, stripeConnectedAccount } = data;
+
+        return {
+            gateway: this.getGatewayIdentifier(),
+            'stripe:version': stripeVersion,
+            'stripe:publishableKey': `${stripePublishableKey}/${stripeConnectedAccount}`,
+        };
+    }
+
+    async processAdditionalAction(
```

Comment:
I believe you missed to cover it with unit tests.

Reply:
Yes, it was in progress, now it's done.